src/AppInstallerCLICore/Workflows/RepairFlow.cpp

```diff
@@ -0,0 +1,554 @@
+    {
+        DWORD repairResult = 0;
+
+        if (repairResult != 0)
+        {
+            const auto installerPackageVersion = context.Get<Execution::Data::InstalledPackageVersion>();
+
+            Logging::Telemetry().LogRepairFailure(
+                installerPackageVersion->GetProperty(PackageVersionProperty::Id),
+                installerPackageVersion->GetProperty(PackageVersionProperty::Version),
+                m_repairType,
+                repairResult);
+
+            // Show log path if available
+            if (context.Contains(Execution::Data::LogPath) && std::filesystem::exists(context.Get<Execution::Data::LogPath>()))
+            {
+                auto installerLogPath = Utility::LocIndString{ context.Get<Execution::Data::LogPath>().u8string() };
+                context.Reporter.Info() << Resource::String::InstallerLogAvailable(installerLogPath) << std::endl;
+            }
+        }
+        else
+        {
+            context.Reporter.Info() << Resource::String::RepairFlowRepairSuccess << std::endl;
+        }
+    }
+
+    void RepairSinglePackage::operator()(Execution::Context& context) const
+    {
+        THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), m_operationType != OperationType::Repair);
+
+        context <<
+            RepairApplicabilityCheck <<
+            GetRepairInfo <<
+            Workflow::ExecuteRepair;
+    }
+
+    void SelectApplicablePackageVersion::operator()(Execution::Context& context) const
+    {
+        auto package = context.Get<Execution::Data::Package>();
+        auto installedPackage = context.Get<Execution::Data::InstalledPackageVersion>();
+        const bool reportVersionNotFound = m_isSinglePackage;
+
+        bool isRepair = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseRepair);
+
+        Utility::Version version;
+        Utility::Version installedVersion;
+
+        if (isRepair)
+        {
+            installedVersion = Utility::Version(installedPackage->GetProperty(PackageVersionProperty::Version));
+        }
+
+        ManifestComparator manifestComparator(context, isRepair ? installedPackage->GetMetadata() : IPackageVersion::Metadata{}); // Use the installed package metadata for comparison if we are repairing
+
+        bool versionFound = false;
+        bool installedTypeInapplicable = false;
+
+        if (isRepair && installedVersion.IsUnknown() && !context.Args.Contains(Execution::Args::Type::IncludeUnknown))
+        {
+            if (reportVersionNotFound)
+            {
+                context.Reporter.Info() << Resource::String::NoApplicableInstallers << std::endl;
+            }
+
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_NO_APPLICABLE_INSTALLER);
+        }
+
+        const auto& versionKeys = package->GetAvailableVersionKeys();
+        bool matchingVersionFound = false;
+
+        for (const auto& versionKey : versionKeys)

```

Comment:
By calling `package->GetAvailableVersion()` with a VersionKey argument, it is possible to avoid a loop. However, this requires an implementation to convert the installed Package version to a version key.

For the time being, I will implement a for loop with a simplified code block inside.